### ISSUE TITLE
How to sync Dolt(Hub) with PostgreSQL?

### ISSUE THREAD

**Issue by steve-scx** [dolthub/dolt#200]
I'm looking to build a data integration pipeline with Dolt sitting between the live data in a production PostgreSQL database, and various local and staging-area PostgreSQL databases used for automated and manual data integration.

How would one be able to synchronize between Dolt and PostgreSQL in such a scenario? Is there direct support for that, or does one have to come up with scripts?

**Comment by oscarbatori** [comment]
@steve-scx good question. Currently the easiest way to do that would be to `dolt export ...`, and then import the file produced. In the medium term we will likely move to something ODBC based, but we haven't totally fleshed that out.

For a higher level solution that doesn't require explicit command line interaction, we have a Python library that uses Pandas under the hood, and thus you could write a couple of lines of Python code as follows:
```
from doltpy.core import Dolt
repo = Dolt('repo_dir')
data = read
data.to_sql(...)
```
Note that you would have to pass and database connector to `to_sql`, but that's pretty straight forward, and you can find the details [here](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_sql.html).

So, in summary, it's possible, and will get easier over time as we refine Dolt to speak more seamlessly with RDBMS solutions. If you're interested in sharing more about your use-case so that we can incorporate it into our thinking on this, that'd be great.

**Comment by steve-scx** [comment]
Interesting, thanks for that.

Here's a bit more info about our use-case:

We have a dataset that presently "lives" in our production Postgres database. Our support team occasionally makes corrections to the dataset directly in production through our homebrew "AdminUI".

We are going to start curating this dataset using a combination of human and machine intelligence, using a combination of local and remote workers, some Mechanical Turk style, others using our AdminUI.

But we don't want any of these curation results to go directly into production, instead, we want them working against an offline revision controlled dataset, which can potentially be merged with the work of other curators and then merged into production.

At the same time, we'd like our support team to be able to continue making changes directly in production, and have those changes be merged back into the revision controlled dataset as well. We're open to creating a staging environment where our support team can work instead, which would be backed by revision control, and then be merged into production.

But given that our dataset and tools live both in Postgres and a revision control system, it seems like we can't avoid solving the problem of how to merge between them in both directions without losing or corrupting data.

How does that fit with the workflows and solution you've been thinking about?

**Comment by oscarbatori** [comment]
Steven - thanks for the detail on your use-case, that's really helpful. We have indeed thought a lot about the workflow you have in mind. I spent much of my career in quantitative finance, and one ever present issue was humans making updates and corrections to production data. Oftentimes it wasn't feasible to build proper controls into the application layer, meaning all kinds of unsavory database interactions were necessary. I very early conceived of Dolt as a mechanism for managing human interactions with business critical production data.

Would you be interested in doing a quick call, and we can share more detail on how this could be achieved? We'd also be grateful to develop a keener understanding of the subtleties of getting the solution right. If so, just shoot me a note at oscar@liquidata.co, and we can discuss from there.

**Comment by oscarbatori** [comment]
@ptman correct, jump in [Discord](https://discord.com/invite/RFwfYpu) if you have any questions.

**Comment by oscarbatori** [comment]
Closing this out
